### ISSUE TITLE
tf.lite.TFLiteConverter crashes when converting Keras model

### ISSUE THREAD

**Issue by haifengkao** [tensorflow/tensorflow#41694]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): OSX 10.15.5- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): v1.12.1-37224-ga6cd18a133 2.4.0-dev20200722
- Python version: 3.7

**Describe the current behavior**
It throws exception
![截圖 2020-07-24 下午3 37 41](https://user-images.githubusercontent.com/4080524/88370104-f6229600-cdc3-11ea-8d7b-4f839cb5ac04.jpg)

**Describe the expected behavior**
It should finish the conversion successfully

**Standalone code to reproduce the issue**
```
import numpy as np
import tensorflow as tf

model = tf.keras.models.Sequential([
    tf.keras.layers.Input(shape=(28, 28), name='input'),
    tf.keras.layers.Bidirectional(
        tf.keras.layers.LSTM(20, return_sequences=True)),
    tf.keras.layers.Flatten(),
    tf.keras.layers.Dense(10, activation=tf.nn.softmax, name='output')
])
model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])
model.summary()

run_model = tf.function(lambda x: model(x))
# This is important, let's fix the input size.
BATCH_SIZE = 1
STEPS = 28
INPUT_SIZE = 28
concrete_func = run_model.get_concrete_function(
    tf.TensorSpec([BATCH_SIZE, STEPS, INPUT_SIZE], model.inputs[0].dtype))

# model directory.
MODEL_DIR = "keras_lstm"

# converter = tf.lite.TFLiteConverter.from_concrete_functions([concrete_func])
# converter = tf.lite.TFLiteConverter.from_saved_model(MODEL_DIR)
converter = tf.lite.TFLiteConverter.from_keras_model(model)
converter.experimental_new_converter = True
tflite_model = converter.convert()

with tf.io.gfile.GFile('tflite_test.tflite', 'wb') as f:
    f.write(tflite_model)
```

**Comment by abattery** [comment]
Renjie, could you take a look? This behavior is reproducible at HEAD and the error is coming from the SplitMergedOperandsPass.

**Comment by abattery** [comment]
The model has a unidirectional_sequence_lstm op, which has the same tensor result from tfl.fill op at both stateful input tensors.

**Comment by amahendrakar** [comment]
Was able to convert the model successfully with [TF v2.2](https://colab.research.google.com/gist/amahendrakar/4269072fef0f59a28396ccefbe4ddef5/41694.ipynb). Facing issue on running the code with [TF v2.3.0-rc2](https://colab.research.google.com/gist/amahendrakar/f51edfffb536854f7a1fb26884dd892f/41694-2-3.ipynb#scrollTo=d8HyaCK3fdbL) and [TF-nightly](https://colab.research.google.com/gist/amahendrakar/b54d7ab3e07b0c821eb6076bf260033a/41694-tf-nightly.ipynb). 

Please find the attached gist. Thanks!

**Comment by renjie-liu** [comment]
Can you try to follow the example here:

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/examples/experimental_new_converter/Keras_LSTM_fusion_Codelab.ipynb

```
run_model = tf.function(lambda x: model(x))
# This is important, let's fix the input size.
BATCH_SIZE = 1
STEPS = 28
INPUT_SIZE = 28
concrete_func = run_model.get_concrete_function(
    tf.TensorSpec([BATCH_SIZE, STEPS, INPUT_SIZE], model.inputs[0].dtype))

# model directory.
MODEL_DIR = "keras_lstm"
model.save(MODEL_DIR, save_format="tf", signatures=concrete_func)

converter = tf.lite.TFLiteConverter.from_saved_model(MODEL_DIR)
tflite_model = converter.convert()
```

Thanks a lot!

Cheers,

**Comment by karimnosseir** [comment]
@haifengkao One other way beside what Renjie mentioned.
is setting the batch size to 1 for the input
like this
tf.keras.layers.Input(shape=(28, 28), name='input', batch_size=1)

So the sample above will be


model = tf.keras.models.Sequential([
    tf.keras.layers.Input(shape=(28, 28), name='input', batch_size=1),
    tf.keras.layers.Bidirectional(
        tf.keras.layers.LSTM(20, return_sequences=True)),
    tf.keras.layers.Flatten(),
    tf.keras.layers.Dense(10, activation=tf.nn.softmax, name='output')
])
model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'],experimental_steps_per_execution=1)
model.summary()

converter = tf.lite.TFLiteConverter.from_keras_model(model)
tflite_model = converter.convert()

with tf.io.gfile.GFile('/tmp/tflite_test.tflite', 'wb') as f:
    f.write(tflite_model)

**Comment by haifengkao** [comment]
@renjie-liu I already tried it 2 days ago, and it works as I stated in the issue https://github.com/tensorflow/tensorflow/issues/41653

**Comment by haifengkao** [comment]
@karimnosseir setting batch size to 1 works as well.

**Comment by renjie-liu** [comment]
Great! Shall we close the bug then?

The reason is in 2.2 the lstm ops are not fused but now they are.

**Comment by haifengkao** [comment]
@renjie-liu  Of course not. You figured out the root cause, but it still crashes. Is fused op supposed to throw exception?

**Comment by renjie-liu** [comment]
Sure, we can through an exception if the batch size is not set during the conversion.

**Comment by haifengkao** [comment]
@renjie-liu True, but the exception message should be "batch size not set" instead of dumping a whole screen of numbers. 
If this concept is foreign to you, you may check [defensive programming](https://enterprisecraftsmanship.com/posts/defensive-programming/)

**Comment by renjie-liu** [comment]
Hi Jaesung,

can you throw a warning in the python layer?

thanks

**Comment by sushreebarsa** [comment]
Was able to replicate the issue with TF v2.5,please find the gist [here](https://colab.research.google.com/gist/sushreebarsa/f68daf4dd3d26b7be396df7db891b308/untitled283.ipynb) ..Thanks!

**Comment by sushreebarsa** [comment]
Closing this issue as it is fixed in latest version of TensorFlow. Please feel free to reopen the issue if you still have a concern. Thanks!
